lib/SILGen/SILGenConstructor.cpp

```diff
@@ -1114,8 +1114,12 @@ void SILGenFunction::emitClassConstructorInitializer(ConstructorDecl *ctor) {
                                    /*ignored parameters*/ 1);
 
   SILType selfTy = getLoweredLoadableType(selfDecl->getTypeInContext());
-  ManagedValue selfArg = B.createInputFunctionArgument(selfTy, selfDecl);
-  
+  LifetimeAnnotation annotation = selfTy.getLifetime(F).isEagerMove()
```

Comment:
I think this needs a comment given that the logic would seem to be backward to a future me re-reading the code.

Reply:
Added

```
  // Force a lexical lifetime for an eagerMove class' init to ensure that the
  // body of its deinit always runs after the body of that init.
```